src/test/java/com/firebolt/jdbc/resultset/FireboltResultSetTest.java

```diff
@@ -1058,15 +1060,47 @@ void shouldGetObjectsForNumericTypes() throws SQLException {
 		inputStream = getInputStreamWithNumericTypes();
 		resultSet = createResultSet(inputStream);
 		resultSet.next();
+		assertEquals((byte)1, resultSet.getObject(1, Byte.class));
+		assertEquals((short)1, resultSet.getObject(1, Short.class));
 		assertEquals(1, resultSet.getObject(1, Integer.class));
+		assertEquals(1L, resultSet.getObject(1, Long.class));
 		assertEquals(new BigInteger("1"), resultSet.getObject(1, BigInteger.class));
 		assertEquals(1, resultSet.getObject(1, Long.class));
+		assertEquals(1, resultSet.getObject(1, Float.class));
+		assertEquals(1, resultSet.getObject(1, Double.class));
+
+		// the number is too big
+		FireboltException e = assertThrows(FireboltException.class, () -> resultSet.getObject(2, Integer.class));
+		assertEquals(TYPE_TRANSFORMATION_ERROR, e.getType());
+		assertEquals(NumberFormatException.class, e.getCause().getClass());
```

Comment:
Let's get more of these negative tests, e.g. int -> short, int -> byte to make sure we're always failing for all these cases.